website/server/libs/payments/apple.js

```diff
@@ -140,27 +136,35 @@ api.subscribe = async function subscribe (sku, user, receipt, headers, nextPayme
     throw new NotAuthorized(api.constants.RESPONSE_NO_ITEM_PURCHASED);
   }
 
-  let transactionId;
+  let originalTransactionId;
+  let newTransactionId;
 
   for (const purchaseData of purchaseDataList) {
     const dateTerminated = new Date(Number(purchaseData.expirationDate));
     if (purchaseData.productId === sku && dateTerminated > new Date()) {
-      transactionId = purchaseData.transactionId;
+      originalTransactionId = purchaseData.originalTransactionId;
+      newTransactionId = purchaseData.transactionId;
       break;
     }
   }
 
-  if (transactionId) {
+  if (originalTransactionId) {
+    if (user && user.isSubscribed() && user.purchased.plan.customId !== originalTransactionId) {
```

Comment:
`customId`? I'm guessing this should be `customerId`?

Reply:
Went ahead and made that change as it was breaking tests all by itself, making it hard to tell what else needs to change!